core/services/workflows/engine.go

```diff
@@ -772,33 +772,21 @@ func (e *Engine) workerForStepRequest(ctx context.Context, msg stepRequest) {
 		l.Errorf("failed to resolve step in workflow; error %v", verr)
 	}
 
-	info, err := curStep.capability.Info(ctx)
-	if err != nil {
-		l.Errorf("failed to get capability info: %s", err)
-	}
-
-	spendLimits := []capabilities.SpendLimit{}
+	spendLimit := decimal.NewNullDecimal(decimal.Zero)
```

Comment:
Something feels weird about some of this metering logic moving into `e.executeStep`, but not all of it.

Reply:
v1 was weird. I'll look over it again and see if I can move some logic out of `e.executeStep`